Tunning checkstyle outside of Gradle

So after realizing that @jandegr probably meant why even use gradle to run checkstyle (that would actually save the time of the initialization of Gradle so also related to #910 ) in that specific check in his comment here: https://github.com/navit-gps/navit/pull/936#issuecomment-549036698 I tried to run checkstyle from the Debian package itself and get an error:
```
$ checkstyle -c checkstyle.xml ./navit/android/src/org/navitproject/navit/NavitAppConfig.java
log4j:WARN No appenders could be found for logger (com.puppycrawl.tools.checkstyle.Main).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - TreeWalker is not allowed as a parent of LineLength Please review 'Parent Module' section for this Check in web documentation if Check is stan
dard.
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:472)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:398)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:326)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:187)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:124)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: TreeWalker is not allowed as a parent of LineLength Please review 'Parent Module' section for this Check in web documentation if Check is standard.
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:147)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:467)
        ... 5 more
Checkstyle ends with 1 errors.
```

I'm going to dig a bit more as to why it works perfectly with gradle and not without.
@metalstrolch is the error that you have on Gentoo the same? (cf. https://github.com/navit-gps/navit/pull/944#issuecomment-551664037)

(Mostly creating the issue to make sure that not 2 people work on that at the same time)

@aerostitch No, not exactly. It gave me the following :
```
checkstyle -c checkstyle.xml ./navit/android/src/org/navitproject/navit/NavitAppConfig.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "METHOD_REF" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.SeparatorWrapCheck
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:432)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:137)
	at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:420)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:358)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:173)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "METHOD_REF" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.SeparatorWrapCheck
	at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:225)
	at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:162)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:137)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:427)
	... 4 more
Checkstyle ends with 1 errors.

```
Of course this may still be a version issue. The METHOD_REF token was added to checkstyle in 2016, at least according to https://github.com/checkstyle/checkstyle
The version Gentoo has is:
```
emerge -pv checkstyle

These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild   R    ] dev-util/checkstyle-7.2::gentoo  USE="-doc -source" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

```
Lates availale seems to be 8.26

--- a/checkstyle.xml
+++ b/checkstyle.xml
@@ -31,6 +31,12 @@
         <property name="eachLine" value="true"/>
     </module>
 
+    <module name="LineLength">
+        <property name="max" value="120"/>
+        <property name="ignorePattern" value="^package.*|^import.*|a href|href|http://|https://|ftp://"/>
+        <property name="severity" value="error"/>
+    </module>
+
     <module name="TreeWalker">
         <module name="OuterTypeFilename"/>
         <module name="IllegalTokenText">
@@ -44,11 +50,6 @@
             <property name="allowEscapesForControlCharacters" value="true"/>
             <property name="allowByTailComment" value="true"/>
             <property name="allowNonPrintableEscapes" value="true"/>
-        </module>
-        <module name="LineLength">
-            <property name="max" value="120"/>
-            <property name="ignorePattern" value="^package.*|^import.*|a href|href|http://|https://|ftp://"/>
-            <property name="severity" value="error"/>
         </module>
         <module name="AvoidStarImport"/>
         <module name="OneTopLevelClass"/>

--- a/checkstyle.xml
+++ b/checkstyle.xml
@@ -259,7 +259,6 @@
             <property name="allowMissingParamTags" value="true"/>
             <property name="allowMissingThrowsTags" value="true"/>
             <property name="allowMissingReturnTag" value="true"/>
-            <property name="minLineCount" value="2"/>
             <property name="allowedAnnotations" value="Override, Test"/>
             <property name="allowThrowsTagsForSubclasses" value="true"/>
         </module>

--- a/docs/development/programming_guidelines.rst
+++ b/docs/development/programming_guidelines.rst
@@ -206,4 +206,8 @@ For the Java code we follow the Google coding conventions from Google Java Style
 This style is enforced by using Checkstyle. Its definition file can be found at the root of the repository:
 `checkstyle.xml <https://github.com/navit-gps/navit/blob/trunk/checkstyle.xml>`_
 
+.. note::
+
+  Checkstyle 8.25 minimum is required if you want to run checkstyle without using gradle.
+
 Please add yourself to the list of authors, if you make a significant change.

--- a/navit/android/build.gradle
+++ b/navit/android/build.gradle
@@ -68,7 +68,7 @@ android {
         }
 
         checkstyle {
-            toolVersion = '8.10'
+            toolVersion = '8.26'
         }
     }
     applicationVariants.all { variant ->

